File: #0713.subarray-product-less-than-k.cpp
class Solution {
public:
    int numSubarrayProductLessThanK(vector<int>& nums, int k) {

        int size = nums.size();
        int left = 0;
        int right = 0;
        long long num = 1LL;
        int result = 0;
        for (int right = 0; right < size; ++right) {
            num *= nums[right];
            while (left < size && num >= k) {
                num /= nums[left];
                ++left;
            }

            if (left <= right) {
                result += (right - left + 1);
            }
        }
        return result;
    }
};

File: #0713.subarray-product-less-than-k.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var numSubarrayProductLessThanK = function(nums, k) {
   let count = 0;
   for (let i = 0; i < nums.length; i++) {
        let num = nums[i];
        let j = i + 1;
        while (num < k) {
            count++;
            num *= nums[j];
            j++;
        }  
    }
    return count;
};
